### DISCUSSION TITLE
GraphQL - possibility to check if a user is admin

### DISCUSSION THREAD

**Discussion by angrotheman** [craftcms/cms#8827] [Ideas]
There is currently no built-in way to verify that a user is an administrator.
My current workaround is to use the actionSessionInfo Controller and append this 1 line of code: ` $return['admin'] = $user->admin;`

I think it would be great to check via Graph-QL if a user is admin

**Comment by andris-sevcenko** [comment]
The reason why this was excluded is because of a potentially sensitive information leak. We'll revisit this once we get to the point where we layer user permissions on top of graphql schema.

**Comment by angrotheman** [comment]
> The reason why this was excluded is because of a potentially sensitive information leak. We'll revisit this once we get to the point where we layer user permissions on top of graphql schema.

Thanks for your reply. Why should be this a sensitive information leak? Everyone can create his own query with a admin request or without. 

Besides the admin request, it would also be great to check the user groups where the user is added to. 
Maybe is this something less sensitive information?

**Comment by narration-sd** [comment]
@angrotheman  Honestly, have you thought about your desire to use a 'public' schema at all?

You realize you are then inviting the entire internet to come and camp on your server...or anyway, the script kiddies who are constantly, second-by-second, probing IPs to discover such open resources, perhaps unknowlingly of what they are providing to the real bandits who 'gave' them these toys.

Unless the entire installation is on a very soundly protected private intranet, one supposes...
